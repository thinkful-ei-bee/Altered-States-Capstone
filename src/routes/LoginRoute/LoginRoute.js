import React, { Component } from "react";
import LoginForm from "../../components/LoginForm/LoginForm";
import Footer from '../../components/Footer/Footer'

class LoginRoute extends Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => {}
    }
  };

  handleLoginSuccess = () => {
    const { location, history } = this.props;
    const destination = (location.state || {}).from || "/";
    history.push(destination);
  };

  render() {
    return (
      <section>
        <h2>Login</h2>
        <LoginForm onLoginSuccess={this.handleLoginSuccess} />
        <Footer/>
      </section>
    );
  }
}

export default LoginRoute;
