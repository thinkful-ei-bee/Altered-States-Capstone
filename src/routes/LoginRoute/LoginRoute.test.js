import React from 'react';
import LoginRoute from './LoginRoute';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

describe('DashboardRoute', () => {

  it('renders the complete page', () => {
    const wrapper = shallow(<LoginRoute />);
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})