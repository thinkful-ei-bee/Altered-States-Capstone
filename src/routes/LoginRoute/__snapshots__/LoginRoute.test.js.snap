// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRoute renders the complete page 1`] = `
<section>
  <LoginForm
    onLoginSuccess={[Function]}
  />
</section>
`;
